<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>BS3008: Computer Aided Drug Discovery - 4&nbsp; PDB Files and PyMOL Visualizations</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./summary.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">PDB Files and PyMOL Visualizations</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">BS3008: Computer Aided Drug Discovery</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/kevfo/BS3008-Notes" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="" title="Share" id="sidebar-tool-dropdown-0" class="sidebar-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi bi-share"></i></a>
    <ul class="dropdown-menu" aria-labelledby="sidebar-tool-dropdown-0">
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
            <i class="bi bi-bi-twitter pe-1"></i>
          Twitter
          </a>
        </li>
        <li>
          <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
            <i class="bi bi-bi-facebook pe-1"></i>
          Facebook
          </a>
        </li>
    </ul>
  <a href="" class="quarto-color-scheme-toggle sidebar-tool" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle sidebar-tool" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true"><strong>PART 1 : LECTURES</strong></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Theoretical Foundations of BS3008</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./wk2.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Force Fields</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./summary.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Summary</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true"><strong>PART 2 : PRACTICAL SESSIONS</strong></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lab1.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">PDB Files and PyMOL Visualizations</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#pdb-files" id="toc-pdb-files" class="nav-link active" data-scroll-target="#pdb-files"><span class="toc-section-number">4.1</span>  PDB Files</a>
  <ul class="collapse">
  <li><a href="#what-is-a-pdb-file" id="toc-what-is-a-pdb-file" class="nav-link" data-scroll-target="#what-is-a-pdb-file"><span class="toc-section-number">4.1.1</span>  What is a PDB File?</a></li>
  <li><a href="#points-to-note" id="toc-points-to-note" class="nav-link" data-scroll-target="#points-to-note"><span class="toc-section-number">4.1.2</span>  Points to Note</a></li>
  <li><a href="#making-a-pdb-file-for-a-water-molecule" id="toc-making-a-pdb-file-for-a-water-molecule" class="nav-link" data-scroll-target="#making-a-pdb-file-for-a-water-molecule"><span class="toc-section-number">4.1.3</span>  Making a PDB File for a Water Molecule</a></li>
  </ul></li>
  <li><a href="#visualizing-molecules-in-pymol" id="toc-visualizing-molecules-in-pymol" class="nav-link" data-scroll-target="#visualizing-molecules-in-pymol"><span class="toc-section-number">4.2</span>  Visualizing Molecules in PyMOL</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/kevfo/BS3008-Notes/edit/main/lab1.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/kevfo/BS3008-Notes/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">PDB Files and PyMOL Visualizations</span></h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p>If you have taken <a href="https://labs.sbs.ntu.edu.sg/obtl/obtl.aspx?sch=sbs&amp;p=BS1005"><em>BS1005: Biochemistry I</em></a>, then the contents of this practical will be relatively straightforward<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</p>
<p>This lab aims to provide a gentle introduction to PDB files and also PyMOL.</p>
<section id="pdb-files" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="pdb-files"><span class="header-section-number">4.1</span> PDB Files</h2>
<p>The first part of the lab examines the structure of a PDB typical PDB file. The later part of this lab then requires you to write your own PDB file for a water molecule (before opening it up in PyMOL to ensure that you have done it correctly.)</p>
<p>The <strong>P</strong>rotein <strong>D</strong>ata <strong>B</strong>ank (i.e., <strong>PDB</strong>) is an initiative by the <a href="https://www.bnl.gov/world/">Brookhaven National Laboratory</a> that stores three-dimensional data of biological molecules (i.e., biomolecules). A considerable amount of information has since been uploaded to the PDB from various fields in Biology, including but not limited to genomics, proteonomics, and structural biology.</p>
<section id="what-is-a-pdb-file" class="level3" data-number="4.1.1">
<h3 data-number="4.1.1" class="anchored" data-anchor-id="what-is-a-pdb-file"><span class="header-section-number">4.1.1</span> What is a PDB File?</h3>
<p>A <strong>PDB file</strong> is a plain text document that contains metainformation about a biomolecule - for instance, its atoms, its side chains, and its amino acid residues.</p>
<p>The PDB also has <a href="https://files.wwpdb.org/pub/pdb/doc/format_descriptions/Format_v33_Letter.pdf">official documentation</a> on the structure of its files that you can view for reference.</p>
</section>
<section id="points-to-note" class="level3" data-number="4.1.2">
<h3 data-number="4.1.2" class="anchored" data-anchor-id="points-to-note"><span class="header-section-number">4.1.2</span> Points to Note</h3>
<p>Although a typical PDB file contains numerous section and record types, BS3008 only focuses on a few of these records (and their important parts):</p>
<ol type="1">
<li><p><strong>The <code>ATOM</code> Record</strong></p>
<p>This is used to define atoms in a PDB file. With this record, the file writer can also define the positions of the atoms (in a three-dimensional space), define side-chain groups, and also define amino acid residues.</p>
<p>Note that all coordinates of length will be in <em>Angstroms</em> in a PDB file.</p></li>
<li><p><strong>B-factors</strong></p>
<p>This is also known as the <strong>temperature factor</strong>. B-factors describe how much an atom’s placement differs from its average values - the higher the B-factor, the higher its displacement.</p>
<p>Areas with high B-factors are usually red (i.e., hot) and vice versa.</p></li>
<li><p><strong>Occupancy</strong></p>
<p>This parameter is associated with different conformations of a biomolecule.</p>
<p>Depending on how many conformations there are available, the occupancy may be less than one (but whatever the case, the values of all occupancices will always add up to one).</p></li>
</ol>
</section>
<section id="making-a-pdb-file-for-a-water-molecule" class="level3" data-number="4.1.3">
<h3 data-number="4.1.3" class="anchored" data-anchor-id="making-a-pdb-file-for-a-water-molecule"><span class="header-section-number">4.1.3</span> Making a PDB File for a Water Molecule</h3>
<p>A water molecule has two hydrogen atoms and one oxygen atom. Its O-H bond length is 0.957 Angstroms and its H-O-H bond angle is <span class="math inline">\(109^\circ\)</span>.</p>
<p>Hence, first set the Oxygen atom to the origin (0, 0, 0) using an ATOM record<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>:</p>
<pre><code>         1         2         3         4         5         6         7
1234567890123456789012345678901234567890123456789012345678901234567890123456789
ATOM     1   O      A    1      0.000   0.000    0.000  1.00 0.00            O</code></pre>
<p>Thereafter, add in the first hydrogen atom at the coordinates (0.957, 0, 0) using another ATOM record. It not necessary to use the aforementioned coordinates - (0, 0.957, 0) and (0, 0, 0.957) also work:</p>
<pre><code>         1         2         3         4         5         6         7
1234567890123456789012345678901234567890123456789012345678901234567890123456789
ATOM     1   O      A    1      0.000   0.000    0.000  1.00 0.00            O
ATOM     2   H      A    1      0.957   0.000    0.000  1.00 0.00            H</code></pre>
<p>Basic trigonometry can then be used to discover the coordinates of the final hydrogen atom. In this case:</p>
<p><span class="math display">\[\begin{align}
  x &amp;= -\cos(71^\circ) * 0.957 \\
  y &amp;= \sin(71^\circ) * 0.957
\end{align}\]</span></p>
<p>Doing so should yield a pair of coordinates <span class="math inline">\((-0.312, 0.905)\)</span>. The final pair of coordinates (with <strong><code>TER</code></strong> and <strong><code>END</code></strong> records) should be:</p>
<pre><code>         1         2         3         4         5         6         7
1234567890123456789012345678901234567890123456789012345678901234567890123456789
ATOM     1   O      A    1      0.000   0.000    0.000  1.00 0.00            O
ATOM     2   H      A    1      0.957   0.000    0.000  1.00 0.00            H
ATOM     3   H      A    1     -0.312   0.957    0.000  1.00 0.00            H
TER      4
END </code></pre>
<p>You can open your PDB file in PyMOL to verify that your PDB file is correct.</p>
</section>
</section>
<section id="visualizing-molecules-in-pymol" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="visualizing-molecules-in-pymol"><span class="header-section-number">4.2</span> Visualizing Molecules in PyMOL</h2>
<p><strong>PyMOL</strong> is an open-source software for visualizing molecules and measuring their various properties. These “molecules” are often stored in a PDB file format (hence the reason why the first portion of the practical deals with PDB files).</p>
<pre><code>         1         2         3         4         5         6         7         8
12345678901234567890123456789012345678901234567890123456789012345678901234567890
ATOM     38  N       A        -33.869   10.617   7.317  1.00 76.26           N
ATOM     39  CA      A        -34.134    9.234   6.904  1.00 74.56           C
ATOM     40  C       A        -33.089    8.813   5.875  1.00 73.03           C
ATOM     41  O       A        -32.729    9.593   4.989  1.00 72.13           O
ATOM     42  CB      A        -35.552    9.101   6.317  1.00 75.48           C
ATOM     43  CG      A        -36.647    9.421   7.338  1.00 77.63           C
ATOM     44  OD1     A        -36.667    8.874   8.452  1.00 81.93           O
ATOM     45  ND2     A        -37.570   10.304   6.961  1.00 75.30           N
ATOM     46  N       A        -32.604    7.582   5.965  1.00 74.10           N
ATOM     47  CA      A        -31.570    7.168   5.019  1.00 77.48           C
ATOM     48  C       A        -32.043    6.300   3.846  1.00 73.71           C
ATOM     49  O       A        -31.538    5.201   3.613  1.00 76.65           O
ATOM     50  CB      A        -30.416    6.504   5.793  1.00 83.44           C
ATOM     51  CG      A        -29.901    7.343   6.934  1.00 90.57           C
ATOM     52  ND1     A        -29.361    8.601   6.754  1.00 92.16           N
ATOM     53  CD2     A        -29.899    7.125   8.275  1.00 92.50           C
ATOM     54  CE1     A        -29.051    9.123   7.930  1.00 91.80           C
ATOM     55  NE2     A        -29.369    8.247   8.871  1.00 92.32           N
ATOM     56  N       A        -33.001    6.831   3.089  1.00 67.38           N
ATOM     57  CA      A        -33.551    6.148   1.924  1.00 59.99           C
ATOM     58  C       A        -33.046    6.785   0.618  1.00 55.78           C
ATOM     59  O       A        -32.178    7.656   0.640  1.00 54.77           O
ATOM     60  CB      A        -35.079    6.191   1.993  1.00 58.20           C
ATOM     61  CG      A        -35.615    7.603   2.156  1.00 55.61           C
ATOM     62  OD1     A        -36.782    7.745   2.593  1.00 55.40           O
ATOM     63  OD2     A        -34.876    8.561   1.843  1.00 55.57           O
ATOM     64  N       A        -33.583    6.349  -0.515  1.00 54.36           N
ATOM     65  CA      A        -33.167    6.876  -1.814  1.00 53.16           C
ATOM     66  C       A        -33.626    8.317  -2.061  1.00 51.53           C
ATOM     67  O       A        -33.041    9.039  -2.874  1.00 50.28           O
ATOM     68  CB      A        -33.673    5.962  -2.929  1.00 56.36           C
ATOM     69  CG      A        -33.144    4.523  -2.821  1.00 59.12           C
ATOM     70  CD      A        -33.329    3.750  -4.125  1.00 57.06           C
ATOM     71  CE      A        -32.718    2.350  -4.022  1.00 60.09           C
ATOM     72  NZ      A        -32.740    1.596  -5.327  1.00 61.46           N
ATOM     73  N       A        -34.685    8.721  -1.368  1.00 48.02           N
ATOM     74  CA      A        -35.197   10.077  -1.478  1.00 42.97           C
ATOM     75  C       A        -34.131   10.992  -0.891  1.00 43.60           C
ATOM     76  O       A        -33.858   12.059  -1.426  1.00 44.56           O
ATOM     77  CB      A        -36.487   10.260  -0.668  1.00 40.66           C
ATOM     78  CG1     A        -37.593    9.371  -1.249  1.00 39.01           C
ATOM     79  CG2     A        -36.885   11.737  -0.658  1.00 39.33           C
ATOM     80  CD1     A        -38.855    9.309  -0.390  1.00 36.41           C</code></pre>
<p>In the second part of the practical, you are given a PDB file (see above) for a polypeptide that is missing its amino acid labels. You will need to open the file using PyMOL, identify the polypeptide’s N and C terminals, and fill in the three-letter code for each <code>ATOM</code> record’s amino acid in columns 18 to 20.</p>


</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>So much so that you don’t need to come if you don’t want to.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>The column numbers on the first two lines have been provided for reference!<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } 
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./summary.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Summary</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->



</body></html>